src/logic/safe/utils/safeVersion.ts

```diff
@@ -86,3 +88,17 @@ export const getSafeVersionInfo = async (safeVersion: string): Promise<SafeVersi
     logError(Errors._606, err.message)
   }
 }
+
+export const isValidMasterCopy = async (
+  chainInfo: ChainInfo,
+  safeAddress: string,
+  safeVersion: string,
+): Promise<boolean> => {
+  const masterCopyAddressFromProxy = await getMasterCopyAddressFromProxyAddress(safeAddress)
+  const { l2, chainId } = chainInfo
+  const useL2ContractVersion = l2 && semverSatisfies(safeVersion, '>=1.3.0')
+  const getDeployment = useL2ContractVersion ? getSafeL2SingletonDeployment : getSafeSingletonDeployment
+  const expectedMasterCopyAddress = getDeployment()?.networkAddresses[chainId]
```

Comment:
There is a bit of overlap with another function that also has some fallbacks in place called `getSafeContractDeployment`. Would it make sense to reuse it here and only let this function handle the address comparison?

Reply:
I exported `getSafeContractDeployment` and reused it.